### ISSUE TITLE
Mosaic Bell Tile

### ISSUE THREAD

**Issue by skleinfeldt** [PSUEducationalEquity/plonetheme.eeq#27]
As a user I can see a highly noticeable tile with a light blue or red background, a bell image floated left and text next to the bell. As a content editor I can choose the color and enter rich text for the tile. 

Note that we will need the SVG for the bell.

**Comment by skleinfeldt** [comment]
Will use fontawesome for icons etc.

**Comment by witekdev** [comment]
This has been implemented. Requires the theme package to be upgraded to version 1007

The tile is called "Alert" (it sounded better than "Bell" in terms of describing what its supposed to do).
It is available from the Advanced menu, sometimes one has to scroll a bit in the Mosaic Layout editor "Insert" Menu:
ie:

![Screenshot from 2021-06-18 09-47-48](https://user-images.githubusercontent.com/883619/122579542-4e562c80-d055-11eb-8321-f0c2779f9542.jpg)

![Screenshot from 2021-06-18 09-45-53 copy](https://user-images.githubusercontent.com/883619/122579561-51e9b380-d055-11eb-96fb-9f349ca82307.jpg)

The edit interface allows to specify either an internal content item which will be used as the url for title, or an external url. Full url is needed ie. ``` https://www.google.com ```
All the fields are optional, I assume both title and description will be specified but the flexibility is there to only have title or only a description. I didn't factor in those edge cases when styling but we can adjust in the future, if needed.

Currently if no link is specified, I disabled the hover action over the title. If needed anchors.

![Screenshot from 2021-06-18 09-46-03](https://user-images.githubusercontent.com/883619/122581580-7d6d9d80-d057-11eb-9111-0435cf376cd8.jpg)

To extend the color options this is quite easy in code:
Here: https://github.com/PSUEducationalEquity/plonetheme.eeq/blob/master/src/plonetheme/eeq/browser/tiles.py#L44-L50
Here: https://github.com/PSUEducationalEquity/plonetheme.eeq/blob/master/src/plonetheme/eeq/browser/tiles.py#L99-L100
and here: https://github.com/PSUEducationalEquity/plonetheme.eeq/blob/master/src/plonetheme/eeq/theme/less/tiles.less#L47-L71

The responsive behavior should be looking good as well.

Let me know your feedback. Thanks!

**Comment by paulrentschler** [comment]
@MBrahosky I created https://beta.equity.psu.edu/web/not-home to test this feature.

**Comment by MBrahosky** [comment]
@paulrentschler Looks good Paul!

**Comment by paulrentschler** [comment]
We would like the bell icon to appear "in it's own column" so that the text is not wrapped around it. Thus how the red one appears, not the blue one.
![image](https://user-images.githubusercontent.com/1778602/123476875-45240d00-d5cb-11eb-99f7-c059f56996a9.png)


Also not sure if Mosaic will allow for this, but is it possible to have publication and expiration dates for these? If it is easy, we would like to do that; if it is a lot of work or impossible, then forget we asked.

**Comment by witekdev** [comment]
@paulrentschler :+1: to bell styling, ill get that fixed up.

Regarding publication and expiration dates, are you implying something like Silvio created for the Banner panel?
Or do you want to display extra information on the tile itself?

**Comment by paulrentschler** [comment]
@witekdev yes, like what Silvio created for the Banner panel.

**Comment by witekdev** [comment]
I've updated the alert tile styling, on master.

Regarding the date range, it looks to be possible. However, I just ran into an issue with Datetime widget and mosaic modal, 
https://github.com/PSUEducationalEquity/plonetheme.eeq/pull/48 
Hoping maybe @alecpm will have an idea how to proceed though :crossed_fingers:

**Comment by witekdev** [comment]
@paulrentschler this has been pushed to master. You now have 3 extra fields available on the tile:

![tile](https://user-images.githubusercontent.com/883619/123974113-a88faf80-d9bc-11eb-86ec-4d94a782f975.jpg)

If the functionality is enabled, the tile will appear 

To help with the editing, if the tile is expired/inactive, it will appear with a grayed out background in the mosaic layout edit mode (Can remove this style but maybe useful to see if you set the dates correctly when editing or if you have more than one etc)

eg:
![inactivetiles](https://user-images.githubusercontent.com/883619/123975376-92ceba00-d9bd-11eb-8e3a-6ee9b62c2df2.jpg)

Other caveats;
If the publication_date is not set, it will be ignored (ie it is possible to only set the expiration date)
Likewise if the expiration_date is not set, it will be ignored. (possible to only publish at a certain date and forget)

eg:
![exponly](https://user-images.githubusercontent.com/883619/123975233-7894dc00-d9bd-11eb-835a-7d89272011c1.jpg)

**Comment by paulrentschler** [comment]
@witekdev Mike and I reviewed the updates and love the grayed out background for expired/inactive tiles! Is the "Use data range" checkbox absolutely necessary since forgetting to check it means that the dates are ignored?

**Comment by witekdev** [comment]
> @witekdev Mike and I reviewed the updates and love the grayed out background for expired/inactive tiles! Is the "Use data range" checkbox absolutely necessary since forgetting to check it means that the dates are ignored?

We can do away with "Use data range" sure, it's just a hard "ON/OFF" switch for the dates, however since you also have the option of not filling in the dates if you want the tile to be permanently on. We can get rid of this checkbox.
Please confirm if you would like that, and I'll make the change :+1:

**Comment by MBrahosky** [comment]
@witekdev I'll defer to Paul, but given we most likely will have additional content editors on the site, it may be best to eliminate confusion and remove the the checkbox.

**Comment by paulrentschler** [comment]
@witekdev yes, since it is not critical to the implementation, let's get rid of the checkbox. Thanks!

**Comment by witekdev** [comment]
@paulrentschler on master

**Comment by witekdev** [comment]
ps. looks like there's a new version of a security fix
Products.PloneHotfix20210518 = 1.5
also updated on master
